app/routes.py

```diff
@@ -1,2 +1,179 @@
-from flask import Blueprint
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, Blueprint, jsonify, Response, make_response
+from datetime import datetime
+import requests
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["POST"])
+def post_tasks():
+    request_body = request.get_json()
+    if "title" in request_body.keys() and "description" in request_body.keys() and "completed_at" in request_body.keys() :
+        task = Task(title=request_body["title"],
+                    description=request_body["description"], 
+                    completed_at=request_body["completed_at"],
+                    )
+        db.session.add(task)
+        db.session.commit()
+        return jsonify({"task": task.api_response()}), 201
+    else:
+        return make_response(
+            {"details": "Invalid data"
+            }
+        ), 400
+
+@tasks_bp.route("", methods=["GET"])
+def get_tasks():
+    sort_param = request.args.get("sort")
+    if sort_param == "asc":
+        tasks = Task.query.order_by(Task.title).all()
+    elif sort_param == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+    tasks_response = [task.api_response() for task in tasks] 
+    return jsonify(tasks_response), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def get_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response(jsonify(None), 404)
```

Comment:
this works just fine! One thing we could do is use another method to do this all at once for us!

```suggestion
    task = Task.query.get_or_404(task_id)
```